src/ShapeCrawler/ShapeCollection/SlideShapes.cs

```diff
@@ -121,18 +125,39 @@ public void AddPicture(Stream imageStream)
         imageCopy.Position = 0;
         imageStream.Position = 0;
         using var skBitmap = SKBitmap.Decode(imageCopy);
-        var xEmu = UnitConverter.HorizontalPixelToEmu(100);
-        var yEmu = UnitConverter.VerticalPixelToEmu(100);
-        var cxEmu = UnitConverter.HorizontalPixelToEmu(skBitmap.Width);
-        var cyEmu = UnitConverter.VerticalPixelToEmu(skBitmap.Height);
 
-        var pPicture = this.CreatePPicture(imageStream, "Picture");
+        if (skBitmap != null)
+        {
+            var xEmu = UnitConverter.HorizontalPixelToEmu(100);
+            var yEmu = UnitConverter.VerticalPixelToEmu(100);
+            var cxEmu = UnitConverter.HorizontalPixelToEmu(skBitmap.Width);
+            var cyEmu = UnitConverter.VerticalPixelToEmu(skBitmap.Height);
+
+            var pPicture = this.CreatePPicture(imageStream, "Picture");
+
+            var transform2D = pPicture.ShapeProperties!.Transform2D!;
+            transform2D.Offset!.X = xEmu;
+            transform2D.Offset!.Y = yEmu;
+            transform2D.Extents!.Cx = cxEmu;
+            transform2D.Extents!.Cy = cyEmu;
+        }
+        else
+        {
+            // Not a bitmap, let's try it as an SVG
+            Svg.SvgDocument? doc = null;
+            try
+            {
+                doc = Svg.SvgDocument.Open<Svg.SvgDocument>(imageStream);
+            }
+            catch
+            {
+                // Neither bitmap nor svg can load this, so that's an error.
+                throw new ArgumentException("Unable to decode image from supplied stream", nameof(imageStream));
```

Comment:
Please use library custom class [SCException](https://github.com/ShapeCrawler/ShapeCrawler/blob/12982b7384a54285edf97fdded6820f6743245e4/src/ShapeCrawler/Exceptions/SCException.cs#L5) for exceptions.